integrations/streamer_test.go

```diff
@@ -5849,6 +5853,56 @@ stream
 	}
 }
 
+func TestStream_AlertKafka(t *testing.T) {
+	var script = `
+stream
+	|from()
+		.measurement('cpu')
+		.where(lambda: "host" == 'serverA')
+		.groupBy('host')
+	|window()
+		.period(10s)
+		.every(10s)
+	|count('value')
+	|alert()
+		.id('kapacitor/{{ .Name }}/{{ index .Tags "host" }}')
+		.info(lambda: "count" > 6.0)
+		.warn(lambda: "count" > 7.0)
+		.crit(lambda: "count" > 8.0)
+		.kafka()
+		.kafka('topic1', 'topic2')
+`
+
+	clock, et, replayErr, tm := testStreamer(t, "TestStream_Alert", script, nil)
+	defer tm.Close()
+	mock_producer := mocks.NewSyncProducer(t, sarama.NewConfig())
+
+	c := kafka.NewConfig()
+	tm.KafkaService = kafka.NewMockService(c, logService.NewLogger("[test_kafka] ", log.LstdFlags), mock_producer)
+
+	// Expect 3 messages due to 2 calls one with 1 topic (default), and one with 2 topics
+	mock_producer.ExpectSendMessageAndSucceed()
+	mock_producer.ExpectSendMessageAndSucceed()
+	mock_producer.ExpectSendMessageAndSucceed()
+
+	t.Errorf("Ruh Ro")
+
+	err := fastForwardTask(clock, et, replayErr, tm, 13*time.Second)
+	if err != nil {
+		t.Error(err)
+	}
+
+	mock_producer.Close()
+
+	// Run again but this time expect an error from the producer
+	expected_error := errors.New("A predictable failure.")
+	mock_producer.ExpectSendMessageAndFail(expected_error)
+	err = fastForwardTask(clock, et, replayErr, tm, 13*time.Second)
```

Comment:
The fastForwardTask function will only work once since it plays all the data in the TestStream_Alert.srpl file against the task. Calling it again would require you to start the task over again using `testStreamer`.


Help me understand the purpose of the second call. Is it so that you can validate that errors are handled correctly? If so you should be able to do that with a single fastForwardTask call as the alert will process each point in the testdata file so you should be able to have one of those calls fail.

If I am missing the point please explain. Overall this is headed in the right direction, glad you have a nice mock object for the service.